stingray/stats.py

```diff
@@ -0,0 +1,492 @@
+                            2 * n_summed_spectra * n) / (n_summed_spectra)
+
+    return retlev
+
+
+def pds_probability(level, ntrial=1, n_summed_spectra=1, n_rebin=1):
+    r"""Give the probability of a given power level in PDS.
+
+    Return the probability of a certain power level in a Power Density
+    Spectrum of nbins bins, normalized a la Leahy (1983), based on
+    the 2-dof :math:`{\chi}^2` statistics, corrected for rebinning (n_rebin)
+    and multiple PDS averaging (n_summed_spectra)
+
+    Parameters
+    ----------
+    level : float or array of floats
+        The power level for which we are calculating the probability
+
+    Other Parameters
+    ----------------
+    ntrial : int
+        The number of *independent* trials (the independent bins of the PDS)
+    n_summed_spectra : int
+        The number of power density spectra that have been averaged to obtain
+        this power level
+    n_rebin : int
+        The number of power density bins that have been averaged to obtain
+        this power level
+
+    Returns
+    -------
+    epsilon : float
+        The probability value(s)
+    """
+
+    epsilon_1 = stats.chi2.sf(level * n_summed_spectra * n_rebin,
+                              2 * n_summed_spectra * n_rebin)
+
+    epsilon = p_multitrial_from_single_trial(epsilon_1, ntrial)
+    return epsilon
+
+
+def pds_detection_level(epsilon=0.01, ntrial=1, n_summed_spectra=1, n_rebin=1):
+    r"""Detection level for a PDS.
+
+    Return the detection level (with probability 1 - epsilon) for a Power
+    Density Spectrum of nbins bins, normalized a la Leahy (1983), based on
+    the 2-dof :math:`{\chi}^2` statistics, corrected for rebinning (n_rebin)
+    and multiple PDS averaging (n_summed_spectra)
+
+    Parameters
+    ----------
+    epsilon : float
+        The single-trial probability value(s)
+
+    Other Parameters
+    ----------------
+    ntrial : int
+        The number of *independent* trials (the independent bins of the PDS)
+    n_summed_spectra : int
+        The number of power density spectra that have been averaged to obtain
+        this power level
+    n_rebin : int
+        The number of power density bins that have been averaged to obtain
+        this power level
+
+    Examples
+    --------
+    >>> np.isclose(pds_detection_level(0.1), 4.6, atol=0.1)
+    True
+    >>> np.allclose(pds_detection_level(0.1, n_rebin=[1]), [4.6], atol=0.1)
+    True
+    """
+    epsilon = p_single_trial_from_p_multitrial(epsilon, ntrial)
+    epsilon = epsilon.astype(np.double)
+    if isinstance(n_rebin, Iterable):
+        retlev = [stats.chi2.isf(epsilon, 2 * n_summed_spectra * r) /
+                  (n_summed_spectra * r) for r in n_rebin]
+        retlev = np.array(retlev)
+    else:
+        r = n_rebin
+        retlev = stats.chi2.isf(epsilon, 2 * n_summed_spectra * r) \
+            / (n_summed_spectra * r)
+    return retlev
+
+
+def classical_pvalue(power, nspec):
+    """
```

Comment:
Add sentence here: "This is stingray's original implementation of the probability distribution for the power spectrum. It is superseded by the implementation in `pds_probability` for practical purposes, but remains here for backwards compatibility and for its educational value as a clear, explicit implementation of the correct probability distribution."